include/tvm/relay/dataflow_pattern.h

```diff
@@ -537,6 +539,41 @@ DFPattern IsTuple(const Array<DFPattern>& fields);
 /*! \brief Syntatic Sugar for creating a TupleGetItemPattern*/
 DFPattern IsTupleGetItem(const DFPattern tuple, int index = -1);
 
+/*! \brief A printer class to print pattern. */
+class DFPatternPrinter : public ReprPrinter {
+ public:
+  std::stringstream string_stream{};
+
+  std::unordered_map<DFPattern, std::pair<size_t, std::string>, ObjectPtrHash, ObjectPtrEqual>
+      memo_{};
+  std::vector<DFPattern> recursed_patterns{};
```

Comment:
Please document what you mean by "recursed_patterns". Probably "recursive_patterns" would be more correct.

Reply:
Documentation added. To be consistent with the printed text, I choose to name this variable to be auxiliary_patterns